lib/services/local/users.go

```diff
@@ -238,17 +244,18 @@ func (s *IdentityService) UpsertUser(user types.User) error {
 		return trace.Wrap(err)
 	}
 	item := backend.Item{
-		Key:     backend.Key(webPrefix, usersPrefix, user.GetName(), paramsPrefix),
-		Value:   value,
-		Expires: user.Expiry(),
-		ID:      user.GetResourceID(),
+		Key:      backend.Key(webPrefix, usersPrefix, user.GetName(), paramsPrefix),
+		Value:    value,
+		Expires:  user.Expiry(),
+		ID:       user.GetResourceID(),
+		Revision: user.GetRevision(),
 	}
 	_, err = s.Put(context.TODO(), item)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 	if auth := user.GetLocalAuth(); auth != nil {
-		if err = s.upsertLocalAuthSecrets(user.GetName(), *auth); err != nil {
+		if err = s.upsertLocalAuthSecrets(context.TODO(), user.GetName(), *auth); err != nil {
```

Comment:
We already use `context.TODO()` in this function, might extract it to `ctx := context.TODO()` like in `CreateUser`